[dotnet] Regression on successive builds (without rebuild / clean)

### Steps to Reproduce

1.  Use `main` HEAD
2. build (make, make install, make system-install)
3. cd tests/dotnet
4. make compare
5. make run-dotnet

### Expected Behavior

Apps starts on a connected device, greeting you with "Hello World!"

### Actual Behavior

```
Build FAILED.

/Users/poupou/git/xamarin/xamarin-macios/builds/downloads/dotnet-sdk-6.0.100-preview.4.21218.6-osx-x64/packs/Microsoft.iOS.Sdk/14.5.100-ci.main.811/tools/msbuild/iOS/Xamarin.iOS.Common.targets(910,3): error MSB6006: "codesign" exited with code 1. [/Users/poupou/git/xamarin/xamarin-macios/tests/dotnet/size-comparison/MySingleView/dotnet/MySingleView.csproj]
    0 Warning(s)
    1 Error(s)
```

### Problem

`make compare` does a `dotnet build` and `make run-dotnet` also does a `dotnet build`

The second build is incorrect. Codesign thinks it's not needed (and is skipped) but the verify steps is run (as it should) and complain the app is **not** signed (and it's not).

If there's a rebuild or clean between both things will work.

Also if you do a 
```
git revert 94196bf767419c6b21346e94672c0aaf1ddad692
```

Then it works again, c.c. @rolfbjarne wrt 94196bf767419c6b21346e94672c0aaf1ddad692

--- a/dotnet/targets/Xamarin.Shared.Sdk.targets
+++ b/dotnet/targets/Xamarin.Shared.Sdk.targets
@@ -352,6 +352,7 @@
 			<!-- Add all the framework files to ResolvedFileToPublish -->
 			<ResolvedFileToPublish Include="@(_FrameworkFilesToPublish)">
 				<RelativePath>%(_FrameworkFilesToPublish._FrameworkPath)\%(_FrameworkFilesToPublish._FrameworkRelativePath)</RelativePath>
+				<CopyToPublishDirectory>PreserveNewest</CopyToPublishDirectory>
 			</ResolvedFileToPublish>
 		</ItemGroup>
 	</Target>
@@ -366,7 +367,9 @@
 			</_DynamicLibraryToPublish>
 
 			<!-- Add all the dynamic libraries to ResolvedFileToPublish -->
-			<ResolvedFileToPublish Include="@(_DynamicLibraryToPublish)" />
+			<ResolvedFileToPublish Include="@(_DynamicLibraryToPublish)">
+				<CopyToPublishDirectory>PreserveNewest</CopyToPublishDirectory>
+			</ResolvedFileToPublish>
 		</ItemGroup>
 	</Target>
 
@@ -545,6 +548,7 @@
 			<!-- copy the aotdata files to the .app -->
 			<ResolvedFileToPublish Include="%(_AssembliesToAOT.AOTData)" >
 				<RelativePath>$([MSBuild]::MakeRelative($(MSBuildProjectDirectory)$(PublishDir),$(_NativeExecutablePublishDir)))\%(_AssembliesToAOT.Filename).aotdata.%(_AssembliesToAOT.Arch)</RelativePath>
+				<CopyToPublishDirectory>PreserveNewest</CopyToPublishDirectory>
 			</ResolvedFileToPublish>
 		</ItemGroup>
 	</Target>
@@ -714,9 +718,10 @@
 
 		<ItemGroup>
 			<!-- Copy the executable from the intermediate directory to the .app -->
-			<ResolvedFileToPublish
-				Include="$(_IntermediateNativeLibraryDir)$(_NativeExecutableName)"
-				RelativePath="$([MSBuild]::MakeRelative($(MSBuildProjectDirectory)$(PublishDir),$(_NativeExecutablePublishDir)))\$(_NativeExecutableName)"/>
+			<ResolvedFileToPublish Include="$(_IntermediateNativeLibraryDir)$(_NativeExecutableName)">
+				<RelativePath>$([MSBuild]::MakeRelative($(MSBuildProjectDirectory)$(PublishDir),$(_NativeExecutablePublishDir)))\$(_NativeExecutableName)</RelativePath>
+				<CopyToPublishDirectory>PreserveNewest</CopyToPublishDirectory>
+			</ResolvedFileToPublish>
 		</ItemGroup>
 	</Target>
 

